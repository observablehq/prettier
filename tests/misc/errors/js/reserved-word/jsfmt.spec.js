run_spec(
  {
    dirname: __dirname,
    snippets: ["class interface {}", 'import interface from "foo";'],
  },
  ["espree", "meriyah"]
);

run_spec(
  {
    dirname: __dirname,
    snippets: [
      'interface = "foo";',
      "interface + 3;",
      "interface();",
      "class interface {}",
      "interface ? true : false;",
      "function interface() {}",
      'import interface from "foo";',
      "interface.foo;",
      "new interface();",
      '(interface, "foo");',
      "void interface;",
      'const interface = "foo";',
    ],
  },
  ["babel"]
);
