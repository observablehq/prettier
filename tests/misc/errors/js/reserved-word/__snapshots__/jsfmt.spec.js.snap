// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippet: #0 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:1)
> 1 | interface = \\"foo\\";
    | ^"
`;

exports[`snippet: #0 [espree] format 1`] = `
"The keyword 'interface' is reserved (1:7)
> 1 | class interface {}
    |       ^"
`;

exports[`snippet: #0 [meriyah] format 1`] = `
"[1:15]: Unexpected strict mode reserved word (1:15)
> 1 | class interface {}
    |               ^"
`;

exports[`snippet: #1 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:1)
> 1 | interface + 3;
    | ^"
`;

exports[`snippet: #1 [espree] format 1`] = `
"The keyword 'interface' is reserved (1:8)
> 1 | import interface from \\"foo\\";
    |        ^"
`;

exports[`snippet: #1 [meriyah] format 1`] = `
"[1:16]: Unexpected strict mode reserved word (1:16)
> 1 | import interface from \\"foo\\";
    |                ^"
`;

exports[`snippet: #2 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:1)
> 1 | interface();
    | ^"
`;

exports[`snippet: #3 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:7)
> 1 | class interface {}
    |       ^"
`;

exports[`snippet: #4 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:1)
> 1 | interface ? true : false;
    | ^"
`;

exports[`snippet: #5 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:10)
> 1 | function interface() {}
    |          ^"
`;

exports[`snippet: #6 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:8)
> 1 | import interface from \\"foo\\";
    |        ^"
`;

exports[`snippet: #7 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:1)
> 1 | interface.foo;
    | ^"
`;

exports[`snippet: #8 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:5)
> 1 | new interface();
    |     ^"
`;

exports[`snippet: #9 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:2)
> 1 | (interface, \\"foo\\");
    |  ^"
`;

exports[`snippet: #10 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:6)
> 1 | void interface;
    |      ^"
`;

exports[`snippet: #11 [babel] format 1`] = `
"Unexpected reserved word 'interface' (1:7)
> 1 | const interface = \\"foo\\";
    |       ^"
`;
